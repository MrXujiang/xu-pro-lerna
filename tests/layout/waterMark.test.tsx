import React from 'react';
import { mount } from 'enzyme';
import { WaterMark } from '@ant-design/pro-layout';
import { waitForComponentToPaint } from '../util';
import { act } from 'react-dom/test-utils';

describe('WaterMark', () => {
  it('test image watermark', async () => {
    let onloadRef: Function | undefined;

    Object.defineProperty(Image.prototype, 'onload', {
      get() {
        // eslint-disable-next-line no-underscore-dangle
        return this._onload;
      },
      set(onload: Function) {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        onloadRef = onload;
        // eslint-disable-next-line no-underscore-dangle
        this._onload = onload;
      },
    });
    const wrapper = mount(
      <WaterMark
        rotate={0}
        image="https://img.alicdn.com/tfs/TB1YM3LpipE_u4jSZKbXXbCUVXa-280-128.png"
      >
        <div style={{ height: 500 }}>123</div>
      </WaterMark>,
    );

    await waitForComponentToPaint(wrapper, 100);
    wrapper.update();
    act(() => {
      onloadRef?.();
    });
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('test text watermark', () => {
    const wrapper = mount(
      <WaterMark content="Trusple">
        <div style={{ height: 500 }} />
      </WaterMark>,
    );
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('test image watermark', async () => {
    const spy = jest.spyOn(global.console, 'error').mockImplementation();
    const createElement = document.createElement.bind(document);
    // @ts-ignore
    document.createElement = (tagName: string) => {
      if (tagName === 'canvas') {
        return {
          setAttribute: () => null,
          getContext: () => null,
          measureText: () => ({}),
        };
      }
      return createElement(tagName);
    };

    const wrapper = mount(
      <WaterMark
        rotate={0}
        image="https://img.alicdn.com/tfs/TB1YM3LpipE_u4jSZKbXXbCUVXa-280-128.png"
      >
        <div style={{ height: 500 }}>123</div>
      </WaterMark>,
    );

    await waitForComponentToPaint(wrapper, 100);
    wrapper.update();
    // @ts-ignore
    expect(console.error.mock.calls).toEqual([['当前环境不支持Canvas']]);
    wrapper.unmount();
    spy.mockRestore();
  });
});
