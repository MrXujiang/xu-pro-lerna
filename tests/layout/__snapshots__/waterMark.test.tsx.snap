// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaterMark test image watermark 1`] = `
<WaterMark
  image="https://img.alicdn.com/tfs/TB1YM3LpipE_u4jSZKbXXbCUVXa-280-128.png"
  rotate={0}
>
  <div
    className="ant-pro-layout-watermark-wrapper"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "height": 500,
        }
      }
    >
      123
    </div>
    <div
      className="ant-pro-layout-watermark"
      style={
        Object {
          "backgroundImage": "url('')",
          "backgroundRepeat": "repeat",
          "backgroundSize": "332px",
          "height": "100%",
          "left": 0,
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": 9,
        }
      }
    />
  </div>
</WaterMark>
`;

exports[`WaterMark test text watermark 1`] = `
<WaterMark
  content="Trusple"
>
  <div
    className="ant-pro-layout-watermark-wrapper"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "height": 500,
        }
      }
    />
    <div
      className="ant-pro-layout-watermark"
      style={
        Object {
          "backgroundImage": "url('data:image/png;base64,00')",
          "backgroundRepeat": "repeat",
          "backgroundSize": "332px",
          "height": "100%",
          "left": 0,
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": 9,
        }
      }
    />
  </div>
</WaterMark>
`;
